plugins/cookie_manager/test/cookies_test.dart

```diff
@@ -139,6 +136,40 @@ void main() {
       expect(response.requestOptions.headers[HttpHeaders.cookieHeader], null);
     });
   });
+
+  test('cookies replacement', () async {
+    final cookies = [
+      Cookie('foo', 'bar')..path = '/',
+      Cookie('a', 'c')..path = '/',
+    ];
+    final previousCookies = [
+      Cookie('foo', 'oldbar')..path = '/',
+      Cookie('d', 'e')..path = '/',
+      Cookie('e', 'f')..path = '/',
+    ];
+    final newCookies = CookieManager.getCookies(
+      [
+        ...previousCookies,
+        ...cookies,
+      ],
+    );
+    expect(newCookies, 'foo=oldbar; d=e; e=f; foo=bar; a=c');
```

Comment:
Yes you are right, it suppose to mimic the scenario with both previous and new cookies exists together.
I have updated the test code again so that the previous cookies will have null path, in order to match the actual scenario.